spec/models/course_enrichment_spec.rb

```diff
@@ -132,60 +132,91 @@
+        let(:course_enrichment) {
+          create(:course_enrichment,
+          about_course: (%w[word] * 400).join(' ') + " exceeeded",
+          interview_process: (%w[word] * 250).join(' ') + " exceeeded",
+          how_school_placements_work: (%w[word] * 350).join(' ') + " exceeeded",
+
+          fee_details: (%w[word] * 250).join(' ') + " exceeeded",
+          salary_details: (%w[word] * 250).join(' ') + " exceeeded",
+          financial_support: (%w[word] * 250).join(' ') + " exceeeded",
+          course: course)
+        }
+
+        context 'fee based course' do
+          let(:course) { create(:course, :with_higher_education) }
+          context 'exceeded word count' do
+            it { should be false }
+            it { should validate_length_of(:about_course) }
+            it { should validate_length_of(:interview_process) }
+            it { should validate_length_of(:how_school_placements_work) }
 
-        context 'exceeded word count' do
-          let(:course_enrichment) {
-            create(:course_enrichment,
-             about_course: (%w[word] * 400).join(' ') + " exceeeded",
-             interview_process: (%w[word] * 250).join(' ') + " exceeeded",
-             how_school_placements_work: (%w[word] * 350).join(' ') + " exceeeded",
-
-             fee_details: (%w[word] * 250).join(' ') + " exceeeded",
-             salary_details: (%w[word] * 250).join(' ') + " exceeeded",
-             financial_support: (%w[word] * 250).join(' ') + " exceeeded")
-          }
-
-          it { should be false }
-          it 'add errors ' do
-            expect(course_enrichment.errors[:about_course]).to match_array ["Reduce the word count for about course"]
-            expect(course_enrichment.errors[:interview_process]).to match_array ["Reduce the word count for interview process"]
-            expect(course_enrichment.errors[:how_school_placements_work]).to match_array ["Reduce the word count for how school placements works"]
-
-            expect(course_enrichment.errors[:fee_details]).to match_array ["Reduce the word count for fee details"]
-            expect(course_enrichment.errors[:salary_details]).to match_array ["Reduce the word count for salary details"]
-            expect(course_enrichment.errors[:financial_support]).to match_array ["Reduce the word count for financial support"]
+            it { should validate_length_of(:fee_details) }
+            it { should validate_length_of(:financial_support) }
+          end
+          context 'no presence' do
+            ['', nil].each do |presence|
+              let(:course_enrichment) {
+                create(:course_enrichment,
+                  fee_uk_eu: presence,
+                  salary_details: presence,
+                  course: course)
+              }
+              it { should be false }
+
+              it 'add errors ' do
+                expect(course_enrichment.errors[:fee_uk_eu]).to match_array ["is not a number", "can't be blank"]
```

Comment:
struggling to do one liner test for these 

```              it { should validate_presence_of(:fee_uk_eu) }```

must be a context breakness

Reply:
Failure/Error: it { should validate_presence_of(:fee_uk_eu) }

     Shoulda::Matchers::ActiveModel::AllowValueMatcher::AttributeDoesNotExistError:
       The matcher attempted to set :fee_uk_eu on the FalseClass to nil, but
       that attribute does not exist.